src/mesh/grids/non_periodic_cube.cpp

```diff
@@ -0,0 +1,66 @@
+#include <deal.II/grid/grid_generator.h>
+#include <deal.II/grid/grid_tools.h>
+
+#include <Sacado.hpp>
+#include "non_periodic_cube.h"
+
+namespace PHiLiP {
+namespace Grids {
+
+template<int dim, typename TriangulationType>
+void non_periodic_cube_flow(
+    TriangulationType& grid,
+    double domain_left,
+    double domain_right,
+    bool colorize,
+    bool shock_tube,
+    bool shu_osher) 
+{
+    dealii::GridGenerator::hyper_cube(grid, domain_left, domain_right, colorize);
+
+    if (shock_tube || shu_osher) {
+        for (auto cell = grid.begin_active(); cell != grid.end(); ++cell) {
+            // Set a dummy boundary ID
+            cell->set_material_id(9002);
+            if (cell == grid.begin_active())
+            {
+                for (unsigned int face = 0; face < dealii::GeometryInfo<dim>::faces_per_cell; ++face) {
+                    if (shu_osher) {
+                        // Set left boundary to Riemann Far Field condition
+                        if (cell->face(face)->at_boundary()) cell->face(face)->set_boundary_id(1004);
+                    }
+                    else {
+                        // Set left boundary to Wall Boundary
+                        if (cell->face(face)->at_boundary()) cell->face(face)->set_boundary_id(1001);
+                    }
+                }
+            }
+            else if (cell == grid.end())
```

Comment:
@ssaishruthi98 , can you please check why the tests pass with the current bug?

Reply:
The way I had it set up previously, I was looping through the faces of the cell and if it was a boundary, I was setting it to a boundary ID. This resulted in both the left and right boundary for all 3 shock tube cases to be set to the value I specified for the left boundary only. 

Also, apparently there is only one cell in the test cases I ran, so it was never really looping through anything to begin with. Since there is only one cell for those three cases, I changed the implementation of this to just set boundary IDs for face(0) and face(1) of the cell instead of looping through cells/faces.